src/main/java/org/opensearch/ad/ratelimit/CheckpointReadWorker.java

```diff
@@ -183,26 +173,29 @@ protected MultiGetRequest toBatchRequest(List<EntityFeatureRequest> toProcess) {
                     final Exception failure = itemResponse.getFailure().getFailure();
                     if (failure instanceof IndexNotFoundException) {
                         for (EntityRequest origRequest : toProcess) {
-                            // submit to cold start queue
+                            // If it is checkpoint index not found exception, I don't
+                            // need to retry as checkpoint read is bound to fail. Just
+                            // send everything to the cold start queue and return.
                             entityColdStartQueue.put(origRequest);
                         }
-                        // If it is checkpoint index not found exception, I don't
-                        // need to retry as checkpoint read is bound to fail. Just
-                        // send everything to the cold start queue and return.
                         return;
-                    } else if (ExceptionUtil.isNotFound(failure)) {
-                        if (notFoundModels == null) {
-                            notFoundModels = new HashSet<>();
-                        }
-                        notFoundModels.add(modelId);
                     } else if (ExceptionUtil.isRetryAble(failure)) {
                         if (retryableRequests == null) {
                             retryableRequests = new HashSet<>();
                         }
                         retryableRequests.add(modelId);
+                    } else if (ExceptionUtil.isOverloaded(failure)) {
+                        LOG.error("too many get AD model checkpoint requests or shard not available");
+                        setCoolDownStart();
                     } else {
                         LOG.info("Unexpected failure", failure);
```

Comment:
For another PR: Should this be logged as an error?

Reply:
yeah, fixed